source/docs/flow_control.md

```diff
@@ -174,9 +176,72 @@ The low watermark path is as follows:
    it calls `readDisable(false)` on the downstream stream to resume data.
 
 # HTTP/2 codec downstream send buffer
-# HTTP/2 network upstream network buffer
 
-TODO(alyssawilk) implement and document.
+On filter creation, all filters have the opportunity to subscribe to downstream
+watermark events sent by the connection manager, and the router filter takes
+advantage of this.  When a particular downstream stream gets backed up, the router filter
+gets notified and can then readDisable() the upstream data source.  The high
+watermark path is as follows:
+
+ * When `ConnectionImpl::StreamImpl::pending_send_data_` has too much data, it calls
+   `ConnectionImpl::StreamImpl::pendingSendBufferHighWatermark()`
+ * `pendingSendBufferHighWatermark()` calls `StreamCallbackHelper::runHighWatermarkCallbacks()`
+ * `runHighWatermarkCallbacks()` results in all subscribers of `Envoy::Http::StreamCallbacks`
+   receiving an `onAboveWriteBufferHighWatermark()` callback.
+ * `ConnectionManagerImpl::ActiveStream::onAboveWriteBufferHighWatermark()` calls
+   `ConnectionManagerImpl::ActiveStream::callHighWatermarkCallbacks()`
+ * `callHighWatermarkCallbacks()` then in turn calls
+    `DownstreamWatermarkCallbacks::onAboveWriteBufferHighWatermark()` for all
+    filters which registered to receive watermark events
+ * `Envoy::Router::Filter` receives `onAboveWriteBufferHighWatermark()` and calls
+   `readDisable(false)` on the upstream request.
+
+The low watermark path is as follows:
+
+ * When `ConnectionImpl::StreamImpl::pending_send_data_` drains, it calls
+   `ConnectionImpl::StreamImpl::pendingSendBufferLowWatermark()`.
+ * `pendingSendBufferLowWatermark()` calls `StreamCallbackHelper::runLowWatermarkCallbacks()`
+ * `runLowWatermarkCallbacks()` results in all subscribers of `Envoy::Http::StreamCallbacks`
+   receiving an `onBelowWriteBufferLowWatermark()` callback.
+ * `ConnectionManagerImpl::ActiveStream::onBelowWriteBufferLowWatermark()` calls
+   `ConnectionManagerImpl::ActiveStream::callLowWatermarkCallbacks()`
+ * `callLowWatermarkCallbacks()` then in turn calls
+    `DownstreamWatermarkCallbacks::onBelowWriteBufferLowWatermark()` for all
+    filters which registered to receive watermark events.
+ * `Envoy::Router::Filter` receives `onBelowWriteBufferLowWatermark()` and calls
+   `readDisable(false)` on the upstream request.
+
+# HTTP/2 network downstream network buffer
+
+When an downstream network connection buffers too much data, it informs the
+connection manager which passes theh high watermark event to all of the streams
+on the connection.  They pass the watermark event to the router, which calls
+`readDisable()` on the upstream streams. The high watermark path is as follows:
+
+ * The downstream `Network::ConnectionImpl::write_buffer_` buffers too much
+   data.  It calls
+   `Network::ConnectionCallbacks::onAboveWriteBufferHighWatermark()`.
+ * `Envoy::Http::Http2::ConnectionImpl::onAboveWriteBufferHighWatermark()`
```

Comment:
I think that past this point (and same below), you can just reference "same flow as above for the codec high watermark"